src/Smarkio/Revisor/FeedbackCollection.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: ivorodrigues
+ * Date: 31/03/17
+ * Time: 12:41
+ */
+
+namespace Smarkio\Revisor;
+
+
+use Smarkio\Exception\InvalidFeedbackException;
+use Smarkio\Exception\InvalidTokenException;
+use Smarkio\Exception\ServiceUnavailableException;
+
+class FeedbackCollection implements \Iterator
+{
+    const API_BASE_URL = 'https://api.smark.io/';
+
+    /**
+     * @var string
+     */
+    protected $token = '';
+
+    /**
+     * @var int
+     */
+    protected $index = 0;
+
+    /**
+     * @var Feedback[]
+     */
+    protected $collection = array();
+
+    /**
+     * FeedbackCollection constructor.
+     *
+     * @param $token
+     * @param null|string|array $data
+     * @throws InvalidFeedbackException
+     */
+    public function __construct($token, $data = null)
+    {
+        $this->token = $token;
+        if ( is_string($data) )
+        {
+            $data = json_decode($data, true);
+        }
+        if ( empty($data) )
+        {
+            $data = array();
+        }
+        if ( !is_array($data) )
+        {
+            return;
+        }
+
+        foreach ($data as $row)
+        {
+            if ( is_array($row) )
+            {
+                $feedback = new Feedback();
+                isset($row['lead_id']) && $feedback->setLeadId($row['lead_id']);
+                isset($row['status']) && $feedback->setStatus($row['status']);
+                isset($row['instant_at']) && $feedback->setInstantAt($row['instant_at']);
+                unset($row['lead_id']);
+                unset($row['status']);
+                unset($row['instant_at']);
+                $feedback->addExtraFields($row);
+
+                $row = $feedback;
+            }
+            if ( $row instanceof Feedback )
+            {
+                array_push($this->collection, $row);
+            }
+            else
+            {
+                throw new InvalidFeedbackException('Unable to generate a Feedback from serialized data.'.var_dump($row, true));
+            }
+        }
+    }
+
+
+    public function current()
```

Comment:
As this code is public we should pay more attention to have PhpDocs even if we define as ``@inheritdoc``